### PR TITLE
dynamic rules fixes and adjustments

### PR DESCRIPTION
**Pull request by martenson**
* do not send hisat2 to stampede with cached index selected
* every rnastar job with need_mb>30G has to go to bridges
* fix punt tools filter to actually prevent jobs to be send to Stampede when they use indexed ref
* refactoring
* more tests

closes https://github.com/galaxyproject/usegalaxy-playbook/issues/214 and https://github.com/galaxyproject/usegalaxy-playbook/issues/215

### CODE REVIEW

File: env/common/files/galaxy/dynamic_rules/test/unit/test_mdw.py [hunk 1]
```diff
@@ -186,7 +198,25 @@
...
+     "sbatch_node": "roundup",
+     "resource_params": {"tacc_compute_resource": "stampede_normal"},
+     "return_submit_native_specification": "--partition=normal --nodes=1 --cpus-per-task=16 --time=48:00:00 --account=TG-MCB140147",
+     "return_destination_id": "stampede_normal"},
+    {"ref_size": 1024 ** 4,
+     "tool": tool_hisat2_history,
+     "sbatch_node": "roundup",
+     "resource_params": {"tacc_compute_resource": "stampede_normal"},
      "return_nativeSpecification": "--partition=multi --nodes=1 --cpus-per-task=6 --time=36:00:00",
      "return_destination_id": "slurm_multi"},
```
Comment (natefoo) [inline env/common/files/galaxy/dynamic_rules/test/unit/test_mdw.py]: These seem backwards to me? If it's indexed it shouldn't return stampede, if it's history it should?

Comment (martenson) [reply env/common/files/galaxy/dynamic_rules/test/unit/test_mdw.py]: for reference: we figured out the actual rule was reversed (thank you unit test!) 
cc52b8b should fix it

**Comment by martenson** [comment]
@natefoo should be ready for another review, I merged commits fixing #215 here to make it easier for myself, please let me know if you prefer them split

**Comment by martenson** [comment]
ping @natefoo

**Comment by martenson** [comment]
congrats but no merge? 😭

**Comment by natefoo** [comment]
Gotta make sure I'm really ready for such a big change.

**Pull request merged**